### ISSUE TITLE
Failure to install rabbitmq admin via curl when proxy set

### ISSUE THREAD

**Issue by madAndroid** [voxpupuli/puppet-rabbitmq#663]
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 5.1.0
- Ruby: ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-linux]
- Distribution: CentOS7
- Module version: 8

## How to reproduce (e.g Puppet code you use)

Manifest invocation in the profile module:
```
class { '::rabbitmq':                     
  admin_enable                => true,    
  delete_guest_user           => true,    
  wipe_db_on_cookie_change    => true,    
  erlang_cookie               => 'cookie',
  config_stomp                => true,    
  repos_ensure                => false,    
  package_ensure              => $version,
}                                         
```

Set `http_proxy`, `https_proxy` and `no_proxy` environment variables

## What are you seeing
```
       Error: Execution of '/bin/curl http://172.17.0.2:15672/cli/rabbitmqadmin -o /tmp/rabbitmqadmin_20171108-146-1iw1csp -fsSL --max-redirs 5 --user guest:guest --insecure' returned 22: curl: (22) The requested URL returned error: 503 Service Unavailable
       Error: /Stage[main]/Rabbitmq::Install::Rabbitmqadmin/Archive[rabbitmqadmin]/ensure: change from 'absent' to 'present' failed: Execution of '/bin/curl http://172.17.0.2:15672/cli/rabbitmqadmin -o /tmp/rabbitmqadmin_20171108-146-1iw1csp -fsSL --max-redirs 5 --user guest:guest --insecure' returned 22: curl: (22) The requested URL returned error: 503 Service Unavailable
```
Curl is not using `--noproxy`, so the command attempts to connect to the proxy, which is not able to handle the request.

## What behaviour did you expect instead

That the curl command uses `--noproxy` to ensure a proxy is not used, even if it's been set as environment variables

**Comment by wyardley** [comment]
How would folks deal with the case where an http proxy _is_ desired? I feel like hard-coding `--noproxy` would be a bad solution.

**Comment by wyardley** [comment]
@madAndroid Does the fix added in #681 help (by allowing arbitrary download options to be passed to the archive resource)?

**Comment by wyardley** [comment]
I believe #681 should resolve this; please re-open or clarify if this doesn't help.
